src/VisualStudio/Core/Impl/Options/AbstractOptionPage.cs

```diff
@@ -110,10 +111,15 @@ public override void SaveSettingsToStorage()
             var oldOptions = s_optionService.GetOptions();
             var newOptions = s_optionStore.GetOptions();
 
-            // Must log the option change before setting the new option values via s_optionService,
-            // otherwise oldOptions and newOptions would be identical and nothing will be logged.
-            OptionLogger.Log(oldOptions, newOptions);
-            s_optionService.SetOptions(newOptions);
+            // Only save options when there have been changes.
+            var haveOptionsChanged = newOptions.GetChangedOptions(oldOptions).Any();
+            if (haveOptionsChanged)
```

Comment:
is this perf optimization or required due to change in this PR?

Reply:
Moving into the OptionService.SetOptions method so that we do not call options changed on the workspace when nothing has changed.